modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -541,6 +552,58 @@ public CodegenModel fromModel(String name, Model model, Map<String, Model> allDe
         return objs;
     }
 
+    public void preprocessSwagger(Swagger swagger) {
+        if (swagger != null && swagger.getPaths() != null) {
+            for (String pathname : swagger.getPaths().keySet()) {
+                Path path = swagger.getPath(pathname);
+                if (path.getOperations() != null) {
+                    for (Operation operation : path.getOperations()) {
+                        boolean hasFormParameters = false;
+                        for (Parameter parameter : operation.getParameters()) {
+                            parameter.getVendorExtensions().put("x-isBody", parameter instanceof BodyParameter);
```

Comment:
Have you tried the mustache tag `isBodyParam` to see if it meets your requirement? If it does, I suggest we avoid introducing another tag `x-isBody` via vendor extensions.

Ref: https://github.com/swagger-api/swagger-codegen/wiki/Mustache-Template-Variables